### ISSUE TITLE
NoSuchMethodError when using a text index and a starts with clause

### ISSUE THREAD

**Issue by InverseIntegral** [neo4j/neo4j#12967]
When I run the following queries:

```
CREATE TEXT INDEX FOR (n:U) ON (n.p)
CREATE (:U { p:"" })
MATCH (n:U) WHERE n.p STARTS WITH "ABC" RETURN COUNT(n)
```

A `NoSuchMethodError` is thrown:

```
Exception in thread "main" java.lang.NoSuchMethodError: 'int org.apache.lucene.search.IndexSearcher.getMaxClauseCount()'
	at org.neo4j.kernel.api.impl.schema.trigram.TrigramQueryFactory.trigramSearch(TrigramQueryFactory.java:91)
	at org.neo4j.kernel.api.impl.schema.trigram.TrigramQueryFactory.stringPrefix(TrigramQueryFactory.java:52)
	at org.neo4j.kernel.api.impl.schema.trigram.TrigramIndexReader.toLuceneQuery(TrigramIndexReader.java:70)
	at org.neo4j.kernel.api.impl.schema.AbstractTextIndexReader.query(AbstractTextIndexReader.java:74)
	at org.neo4j.kernel.impl.newapi.Read.nodeIndexSeek(Read.java:119)
	at org.neo4j.cypher.internal.runtime.interpreted.TransactionBoundReadQueryContext.innerNodeIndexSeek(TransactionBoundQueryContext.scala:792)
	at org.neo4j.cypher.internal.runtime.interpreted.TransactionBoundReadQueryContext.nodeIndexSeek(TransactionBoundQueryContext.scala:676)
	at org.neo4j.cypher.internal.planning.ExceptionTranslatingReadQueryContext.nodeIndexSeek(ExceptionTranslatingQueryContext.scala:158)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.EntityIndexSeeker.indexSeek(EntityIndexSeeker.scala:92)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.EntityIndexSeeker.indexSeek$(EntityIndexSeeker.scala:80)
	at org.neo4j.cypher.internal.runtime.slotted.pipes.NodeIndexSeekSlottedPipe.indexSeek(NodeIndexSeekSlottedPipe.scala:37)
	at org.neo4j.cypher.internal.runtime.slotted.pipes.NodeIndexSeekSlottedPipe.internalCreateResults(NodeIndexSeekSlottedPipe.scala:62)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.Pipe.createResults(Pipe.scala:43)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.Pipe.createResults$(Pipe.scala:40)
	at org.neo4j.cypher.internal.runtime.slotted.pipes.NodeIndexSeekSlottedPipe.createResults(NodeIndexSeekSlottedPipe.scala:37)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.PipeWithSource.computeDecoratedResult(Pipe.scala:91)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.PipeWithSource.createResults(Pipe.scala:81)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.PipeWithSource.computeDecoratedResult(Pipe.scala:91)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.PipeWithSource.createResults(Pipe.scala:81)
	at org.neo4j.cypher.internal.runtime.interpreted.PipeExecutionResult.request(PipeExecutionResult.scala:73)
	at org.neo4j.cypher.internal.result.StandardInternalExecutionResult.request(StandardInternalExecutionResult.scala:90)
	at org.neo4j.cypher.internal.result.ClosingExecutionResult.request(ClosingExecutionResult.scala:149)
	at org.neo4j.cypher.internal.javacompat.ResultSubscriber.fetchResults(ResultSubscriber.java:296)
	at org.neo4j.cypher.internal.javacompat.ResultSubscriber.nextFromSubscriber(ResultSubscriber.java:272)
	at org.neo4j.cypher.internal.javacompat.ResultSubscriber.fetchNextOrNull(ResultSubscriber.java:249)
	at org.neo4j.cypher.internal.javacompat.ResultSubscriber.fetchNextOrNull(ResultSubscriber.java:57)
	at org.neo4j.internal.helpers.collection.PrefetchingIterator.peek(PrefetchingIterator.java:56)
	at org.neo4j.internal.helpers.collection.PrefetchingIterator.hasNext(PrefetchingIterator.java:44)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:132)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
```

**Neo4j Version:** 5.1.0
**Operating System:** 6.0.2 Arch Linux
**API:** Java

### Expected behavior
I would expect the queries not to thrown any exception.

### Actual behavior
An exception is thrown when executing the last query.

**Comment by MishaDemianenko** [comment]
Hi, @InverseIntegral 
would it be possible to share your `debug.log` or your classpath with us for investigation? 
Thx in advance!

**Comment by InverseIntegral** [comment]
Hey @MishaDemianenko,
I think the issue was that a different dependeny (janusgraph) included lucene 8.9.0 whereas neo4j uses 9.3.0.
Removing said dependency fixed the problem.
